File: React_404/cvs-app/src/service/review.service.js
import axios from "axios";
import { BASE_API_URL } from "../common/constants";

const API_URL=BASE_API_URL+"/review";

class ReviewService {
    saveReview(review, p_id, mid) {
        return axios.post(`${API_URL}/add/${p_id}/${mid}`, review);
    }
    deleteReview(rno) {
        return axios.delete(API_URL + "/" + rno);
    }
    getAllReviews() {
        return axios.get(API_URL + "/list");
    }
    getReviewById(rno) {
        return axios.get(API_URL + "/read" + "/" + rno);
    }
    modifyReview(rno, review) {
        return axios.post(API_URL + "/modify" + "/" + rno, review);
    }
}

const reviewService = new ReviewService();
export default reviewService;

File: SpringBoot_404/demo/src/main/java/com/example/demo/review/service/ReviewService.java
package com.example.demo.review.service;

import com.example.demo.productAdmin.dto.ProductDTO;
import com.example.demo.review.domain.Review;
import com.example.demo.review.domain.ReviewImage;
import com.example.demo.review.dto.ReviewDTO;

import java.util.List;

public interface ReviewService {

    ReviewDTO saveReview(ReviewDTO reviewDTO, Long p_id, String mid);

    void deleteReview(Long rno);

    List<Review> findAllReviews();

    Review findReviewById(Long rno);

    Review modifyReview(Review review);

}
